packages/grid/data-grid/src/tests/pagination.DataGrid.test.tsx

```diff
@@ -320,6 +320,56 @@ describe('<DataGrid /> - Pagination', () => {
         );
         expect(getColumnValues(0)).to.deep.equal(['7', '8']);
       });
+
+      it.only('should update the amount of rows rendered and call onPageSizeChange when changing the table height', async () => {
+        const onPageSizeChange = spy();
+
+        const TestCaseAutoPageSize = (props: { nbRows: number; height?: number }) => {
+          const data = useData(props.nbRows, 10);
+
+          return (
+            <div style={{ width: 300, height: props.height }}>
+              <DataGrid
+                columns={data.columns}
+                rows={data.rows}
+                autoPageSize
+                pagination
+                onPageSizeChange={onPageSizeChange}
+              />
+            </div>
+          );
+        };
+        const nbRows = 27;
+        const heightBefore = 780;
+        const heightAfter = 360;
+
+        const { setProps } = render(<TestCaseAutoPageSize nbRows={nbRows} height={heightBefore} />);
+
+        const footerHeight = document.querySelector('.MuiDataGrid-footerContainer')!.clientHeight;
+        const expectedViewportRowsLengthBefore = Math.floor(
+          (heightBefore - DEFAULT_GRID_OPTIONS.headerHeight - footerHeight) /
+            DEFAULT_GRID_OPTIONS.rowHeight,
+        );
+        const expectedViewportRowsLengthAfter = Math.floor(
+          (heightAfter - DEFAULT_GRID_OPTIONS.headerHeight - footerHeight) /
+            DEFAULT_GRID_OPTIONS.rowHeight,
+        );
+
+        let rows = document.querySelectorAll('.MuiDataGrid-viewport [role="row"]');
+        expect(rows.length).to.equal(expectedViewportRowsLengthBefore);
+
+        setProps({ height: heightAfter });
+
+        await raf();
+        await sleep(1000)
```

Comment:
@dtassone did you have trouble with Safari when testing something that requires a redraw of the table ?
If I just wait for the next animation frame, it works on Chrome and Edge but Safari does not yet have the new value (the `useEffect` that publish the events is called after the `await raf()` finishes.
But if I wait for an additional second it does work.